### PR TITLE
refactor(console): unnamed translation

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

Extract the `UnnamedTrans` component logic into util, for this functionality will not be only used as react component.
e.g. display the connector name in the confirm modal. Future feature: LOG-3073:
image

## Linear Issue Reference

N/A

## Testing

Tested in the Admin Console.

### CODE REVIEW

File: packages/console/src/utilities/translation.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import i18next from 'i18next';
+
+export const translateUnnamed = (resource: Record<string, string>) => {
+  const { languages } = i18next;
```
Comment (gao-sun) [inline packages/console/src/utilities/translation.ts:4]: any difference between accessing i18next here and `useTranslation`?

File: packages/console/src/utilities/translation.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  const matchedLanguage = languages.find((language) => resource[language]);
+
+  // Note: the `Unknown` string will not be returned in theory.
+  return (matchedLanguage && resource[matchedLanguage]) ?? 'Unknown';
```
Comment (gao-sun) [inline packages/console/src/utilities/translation.ts:8]: just return undefined?

File: packages/console/src/utilities/translation.ts (hunk 1 above)
Comment (xiaoyijun) [reply packages/console/src/utilities/translation.ts:4]: Yes, `useTranslation` can only be used in react component, and it doesn't work when we want to construct a string with the translated result.

File: packages/console/src/utilities/translation.ts (hunk 2 above)
Comment (xiaoyijun) [reply packages/console/src/utilities/translation.ts:8]: I want to keep it returning `Unknown`, for we don't need to fall back on the return value whenever we use this method and want to get a `string!` value, and also, this method will return `string!` type value in theory.
e.g.:
```js
const result: string = translateUnnamed(resource) ?? 'Unknown';
```

File: packages/console/src/utilities/translation.ts (hunk 1 above)
Comment (xiaoyijun) [reply packages/console/src/utilities/translation.ts:4]: e.g.:
```js
const modalContent = `You're deleting the ${translateUnnamed(resource)} social connector. Are you sure?`;
modal.display(modalContent);
```

**Comment by simeng-li** [comment]
image
typo?

File: packages/console/src/utilities/translation.ts (hunk 2 above)
Comment (simeng-li) [reply packages/console/src/utilities/translation.ts:8]: I would prefer either to return null as before or return some raw data from the resource. i.e. first non-empty key.  'Unknow' makes no difference compared with null, as the user won't get any info from it.  Besides. some zh chars + 'Unknown' is weird as well;

File: packages/console/src/utilities/translation.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import i18next from 'i18next';
+
+export const translateUnnamed = (resource: Record<string, string>) => {
+  const { languages } = i18next;
+  const matchedLanguage = languages.find((language) => resource[language]);
```
Comment (simeng-li) [inline packages/console/src/utilities/translation.ts:5]: matchedTranslation better? as it if returning the actually content not a language.

File: packages/console/src/utilities/translation.ts (hunk 2 above)
Comment (simeng-li) [reply packages/console/src/utilities/translation.ts:8]: Or just return undefined, and let the caller handle the fallback logic.  i.e. if no connector name found, return the connector target.

**Comment by xiaoyijun** [comment]
> image
> 
> typo?

@simeng-li This is a typo in the design, not the code.

File: packages/console/src/utilities/translation.ts (hunk 2 above)
Comment (xiaoyijun) [reply packages/console/src/utilities/translation.ts:8]: Make sense.

**Comment by wangsijie** [comment]
But the confirm modal IS a react component?

File: packages/console/src/utilities/translation.ts (hunk 2 above)
Comment (xiaoyijun) [reply packages/console/src/utilities/translation.ts]: updated.

File: packages/console/src/utilities/translation.ts (hunk 3 above)
Comment (xiaoyijun) [reply packages/console/src/utilities/translation.ts]: updated.

**Comment by xiaoyijun** [comment]
> But the confirm modal IS a react component?

@wangsijie 
We need to pass it to i18n props:
```js
const message = t('some_key', { name: translateUnnamed(resources) });
```

**Review by wangsijie — CHANGES_REQUESTED** [review]
Still don't get it, the original component can meet current demand.

**Pull request closed without merging**